Sources/Amplitude/Amplitude.m

```diff
@@ -1074,19 +1073,37 @@ - (void)enterForeground {
 #endif
     [self runOnBackgroundQueue:^{
         // Fetch the data ingestion endpoint based on current device's geo location.
-        
         [self refreshDynamicConfig];
         [self startOrContinueSessionNSNumber:now];
         self->_inForeground = YES;
-        self->_eventsBuffer = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultEventsFile:self.instanceName]];
-        self->_identifyBuffer = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultIdentifyFile:self.instanceName]];
         if ([self->_eventsBuffer count] > 0 || [self->_identifyBuffer count] > 0) {
             [self uploadEvents];
-            
         }
     }];
 }
 
+- (void)mergeBufferAndFileStorage {
+    NSMutableArray *mergedEvent = [AMPStorage getEventsFromDisk:[AMPStorage getDefaultEventsFile:self.instanceName]];
+    for (NSDictionary *event in self->_eventsBuffer) {
+        long long currentSequenceNumber = [[event objectForKey:@"sequence_number"] longLongValue];
```

Comment:
ah, earlier I mean the merge part could be extracted to a function and used for both..

Reply:
used for both eventsBuffer merge and IdentifyBuffer merge?

Reply:
yeah, i wonder if it could be just taking two arrays and return one merged array

Reply:
Could be a function take two arrays to merge?